docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,300 @@
+   hello world
+   >>>
+
+Build for the stm32 port
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Like the unix port, you need to install some required dependencies
+as detailed in the required dependencies section, then build:
+
+.. code-block:: console
+
+   $ cd ports/stm32
+   $ make submodules
+   $ make
+
+Please refer to the `stm32 documentation <https://github.com/micropython/micropython/tree/master/ports/stm32>`_ 
+for more details on flashing the firmare.
+
+Build for the Windows port
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Windows port is experimental and based on the Unix port. Build on 
+debian/linux systems using the following commands:
+
+.. code-block:: console
+
+   $ cd ports/windows
+   $ sudo apt-get install gcc-mingw-w64
+   $ make CROSS_COMPILE=i686-w64-mingw32-
+
+Then run easily with ``wine``:
+
+.. code-block:: console
+
+   $ wineconsole --backend=curses ./micropython.exe
+
+Refer to the `windows port documentation <https://github.com/micropython/micropython/tree/master/ports/windows>`_
+on more details on building using Cygwin and MS visual Studio 2013 or higher.
+
+Build the documentation
+------------------------
+
+MicroPython documentation is created using ``Sphinx``. If you have already
+installed Python, then install ``Sphinx`` using ``pip``:
+
+.. code-block:: console
+
+   $ pip install sphinx
+
+Navigate to the ``docs`` directory:
+
+.. code-block:: console
+
+   $ cd docs
+
+Build the docs:
+
+.. code-block:: console
+
+   $ make html
+
+Open ``docs/build/html/index.html`` in your browser to view the docs locally. Refer to the 
+documentation on `importing your documentation to use Read the Docs
+<https://docs.readthedocs.io/en/stable/intro/import-guide.html>`_.
+
+Run the tests
+-------------
+
+To run all tests in the testsuite, use ``make``:
+
+.. code-block:: console
+
+   $ make test
+
+.. warning::
+   Make sure you are in the port directory where MicroPython was built before running the tests. 
+   Running the tests outside your port will not work.
+
+Folder structure
+----------------
+
+There are a couple of directories to take note of in terms of where certain implementation details
+are. The following is a break down of the top-level folders in the source code.
+
+``py``
+
+Contains the the compiler, runtime and core library implementation.
+
+``mpy-cross``
+
+Has the MicroPython cross-compiler which pre-compiles the Python scripts to bytecode.
+
+``ports``
+
+Code for all the versions of MicroPython for the supported ports.
+
+``lib``
+
+Low-level C libraries used by any port.
+
+``drivers``
+
+Has drivers for specific hardware and intended to work across multiple ports.
+
+``extmod``
+
+Contains a C implementation of more non-core modules.
+
+``docs``
+
+Has the standard documentation found at https://docs.micropython.org/.
+
+``tests``
+
+An implementation of the test suite.
+
+``tools``
+
+Contains helper tools including the ``upip`` and the ``pyboard.py`` module.
+
+``examples``
+
+Example Python scripts.
```

Comment:
The examples are not strictly Python. There are examples for building MicroPython as a library as well as native modules.